lib/verify/response.rb

```diff
@@ -76,19 +77,20 @@ def address_missing?
 
     def full_name
       first_name = most_recent_verified_value(attributes.fetch("firstNames"))
-      middle_name = most_recent_verified_value(attributes.fetch("middleNames"))
+      middle_name = most_recent_verified_value(attributes.fetch("middleNames"), required: false)
       surname = most_recent_verified_value(attributes.fetch("surnames"))
 
       [first_name, middle_name, surname].compact.join(" ")
     end
 
-    def most_recent_verified_value(attributes)
+    def most_recent_verified_value(attributes, required: true)
       return if attributes.blank?
 
-      attributes.sort_by { |attribute| attribute["from"].present? ? Date.strptime(attribute["from"], "%Y-%m-%d") : 0 }
+      most_recent_verified_attribute = attributes.sort_by { |attribute| attribute["from"].present? ? Date.strptime(attribute["from"], "%Y-%m-%d") : 0 }
         .reverse
         .find { |attribute| attribute["verified"] }
-        .fetch("value")
+      raise MissingResponseAttribute, "No verified value found" if required && most_recent_verified_attribute.nil?
```

Comment:
Ah, can we include the attribute name in the message her to aid debugging?

Reply:
Yeah, I'll have to pass it as another argument as that method only currently receives the attribute hash 👍

Reply:
I guess we should be able to figure it out from the backtrace...

Reply:
I've added a commit that does it, however it feels a bit messy. What do you think?

Reply:
I prefer this version where we pass in the attribute name, rather than the hash!